core/src/main/kotlin/com/malinskiy/marathon/report/junit/JUnitWriter.kt

```diff
@@ -22,7 +22,26 @@ class JUnitWriter(
 
     fun testFinished(devicePoolId: DevicePoolId, device: DeviceInfo, testResult: TestResult, testSummary: TestSummary?) {
         val file = fileManager.createFile(fileType, devicePoolId, device, testResult.test)
-        file.createNewFile()
+
+        try {
+            file.createNewFile()
+        } catch (error: Throwable) {
```

Comment:
if the exception is not related to "file name is too long" we will end up in a loop till the name become empty and still won't be able to create a file